### PR TITLE
Boolean/categorical templates

### PR DESCRIPTION
**Pull request by AdrianM0**
First dataset ready, thought I'd open a PR, so we track changes.

### CODE REVIEW

File: data/tabular/skin_reaction/meta.yaml [hunk 1]
```diff
@@ -18,6 +18,8 @@ targets:
           - verb: causes skin reaction
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:18]:
```suggestion
          - verb: causes a skin reaction
```

File: data/tabular/skin_reaction/meta.yaml [hunk 2]
```diff
@@ -18,6 +18,8 @@ targets:
...
           - verb: causes skin sensitization
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:19]:
```suggestion
          - verb: causes a skin sensitization
```

File: data/tabular/skin_reaction/meta.yaml [hunk 3]
```diff
@@ -18,6 +18,8 @@ targets:
...
           - verb: causes drug induced skin immune reaction
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:20]:
```suggestion
          - verb: causes a drug induced skin immune reaction
```

File: data/tabular/skin_reaction/meta.yaml [hunk 4]
```diff
@@ -18,6 +18,8 @@ targets:
...
+          - adjective: reactive in contact with the skin
+          - gerund: causing skin reaction
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:22]:
```suggestion
          - gerund: causing a skin reaction
```

File: data/tabular/skin_reaction/meta.yaml [hunk 5]
```diff
@@ -59,3 +61,86 @@ bibtex:
       title = {Predicting chemically-induced skin reactions. Part I: QSAR models of skin sensitization
       and their application to identify potentially hazardous compounds},
       journal = {Toxicology and Applied Pharmacology}
+templates:
+    - The {#molecule |!}{SMILES__description} {SMILES#} causes {skin_reaction#no &NULL}reaction with the skin.
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:65]: Why not using the nouns?
```suggestion
    - The {#molecule |!}{SMILES__description} {SMILES#} causes {skin_reaction#no &NULL}{skin_reaction__names__noun}.
```

File: data/tabular/skin_reaction/meta.yaml [hunk 6]
```diff
@@ -59,3 +61,86 @@ bibtex:
...
+    - The molecule with the {SMILES__description} {#representation of |!}{SMILES#} is not {skin_reaction#no &NULL}{skin_reaction__names__adjective}.
+    - Based on the {SMILES__description} {#representation |!}{SMILES#}, there is {skin_reaction#no &NULL}reaction in contact with the skin.
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:67]:
```suggestion
    - Based on the {SMILES__description} {#representation |!}{SMILES#}, there is {skin_reaction#no &NULL}{skin_reaction__names__noun.
```

File: data/tabular/skin_reaction/meta.yaml [hunk 7]
```diff
@@ -59,3 +61,86 @@ bibtex:
       title = {Predicting chemically-induced skin reactions. Part I: QSAR models of skin sensitization
       and their application to identify potentially hazardous compounds},
       journal = {Toxicology and Applied Pharmacology}
+templates:
+    - The {#molecule |!}{SMILES__description} {SMILES#} causes {skin_reaction#no &NULL}reaction with the skin.
+    - The molecule with the {SMILES__description} {#representation of |!}{SMILES#} is not {skin_reaction#no &NULL}{skin_reaction__names__adjective}.
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:66]: Needs fix, uses "not" and recoding: `not {skin_reaction#no &NULL}`

File: data/tabular/skin_reaction/meta.yaml [hunk 8]
```diff
@@ -18,6 +18,8 @@ targets:
           - verb: causes skin reaction
           - verb: causes skin sensitization
           - verb: causes drug induced skin immune reaction
+          - adjective: reactive in contact with the skin
```
Comment (MicPie) [inline data/tabular/skin_reaction/meta.yaml:21]: "reactive in contact" sounds not too common for me?
Maybe switch below to the nouns more often?

File: data/tabular/sr_are_tox21/transform.py [hunk 9]
```diff
@@ -69,6 +70,7 @@ def get_and_transform_data():
                     {"adjective": "toxic in the SR-ARE assay"},
```
Comment (MicPie) [inline data/tabular/sr_are_tox21/transform.py:70]: Is "SR-ARE" really "SR-Antioxidant response element toxicity"? Couldn't find anything with a quick web search.

File: data/tabular/sr_atad5_tox21/transform.py [hunk 10]
```diff
@@ -77,6 +78,7 @@ def get_and_transform_data():
                         "adjective": "toxic in the Luciferase-tagged ATAD5 in human embryonic kidney cells assay"
                     },
                     {"adjective": "toxic in the Luciferase-tagged ATAD5 assay"},
+                    {"gerund": f"showing {target_subfolder} toxicity"},
```
Comment (MicPie) [inline data/tabular/sr_atad5_tox21/transform.py:81]: `target_subfolder` is `SR-ATAD5`, right?

Comment (MicPie) [reply data/tabular/sr_atad5_tox21/transform.py:81]: Maybe add more different gerund?

File: data/tabular/sr_hse_tox21/transform.py [hunk 11]
```diff
@@ -65,6 +66,7 @@ def get_and_transform_data():
                     {"adjective": f"toxic in the {target_subfolder} assay"},
                     {"adjective": "toxic in the SR-Heat shock response assay"},
                     {"adjective": "toxic in the Heat shock response assay"},
+                    {"gerund": f"showing {target_subfolder} toxicity"},
```
Comment (MicPie) [inline data/tabular/sr_hse_tox21/transform.py:69]: Same as above.

File: data/tabular/sr_mmp_tox21/transform.py [hunk 12]
```diff
@@ -73,6 +74,7 @@ def get_and_transform_data():
                     {
                         "adjective": "toxic in the Mitochondrial membrane potential assay"
                     },
+                    {"gerund": f"showing {target_subfolder} toxicity"},
```
Comment (MicPie) [inline data/tabular/sr_mmp_tox21/transform.py:77]: Same as above.

File: data/tabular/sr_atad5_tox21/transform.py (hunk 10 above)
Comment (MicPie) [reply data/tabular/sr_atad5_tox21/transform.py:81]: Hmm, but it is only used one time, so not super important.

File: data/tabular/sr_p53_tox21/transform.py [hunk 13]
```diff
@@ -65,6 +67,7 @@ def get_and_transform_data():
                     {"adjective": f"toxic in the {target_subfolder} assay"},
                     {"adjective": "toxic in the SR-p53 response assay"},
                     {"adjective": "toxic in the p53 response assay"},
+                    {"gerund": f"toxic activity in the {target_subfolder} assay"},
```
Comment (MicPie) [inline data/tabular/sr_p53_tox21/transform.py:70]: Like above?
```suggestion
                    {"gerund": f"showing {target_subfolder} toxicity"},
```

File: data/tabular/tyrosyl-dna_phosphodiesterase_butkiewicz/transform.py [hunk 14]
```diff
@@ -56,11 +56,11 @@ def get_and_transform_data():
                 "type": "boolean",
                 "names": [
                     {"noun": "a tyrosyl-DNA phosphodiesterase 1 (TDP1) inhibitor"},
-                    {"noun": "a tyrosyl-DNA phosphodiesterase 1 inhibitor"},
+                    {"noun": "an inhibitor of tyrosyl-DNA phosphodiesterase 1"},
                     {
-                        "gerund": "inhibiting human tyrosyl-DNA phosphodiesterase 1 (TDP1)"
+                        "gerund": "inhibiting properties of human tyrosyl-DNA phosphodiesterase 1 (TDP1)"
```
Comment (MicPie) [inline data/tabular/tyrosyl-dna_phosphodiesterase_butkiewicz/transform.py:61]:
```suggestion
                        "gerund": "inhibiting the human tyrosyl-DNA phosphodiesterase 1 (TDP1)"
```

File: data/tabular/tyrosyl-dna_phosphodiesterase_butkiewicz/transform.py [hunk 15]
```diff
@@ -56,11 +56,11 @@ def get_and_transform_data():
...
                     },
-                    {"gerund": "inhibiting human tyrosyl-DNA phosphodiesterase 1"},
+                    {"gerund": "inhibiting of human tyrosyl-DNA phosphodiesterase 1"},
```
Comment (MicPie) [inline data/tabular/tyrosyl-dna_phosphodiesterase_butkiewicz/transform.py:63]: Can be removed with the change above?

**Review by MicPie — CHANGES_REQUESTED** [review]

File: data/tabular/peptides_hemolytic/transform.py [hunk 16]
```diff
@@ -145,9 +146,81 @@ def get_and_transform_data():
 pages = {513--530},
 author = {Zhenqin Wu and Bharath Ramsundar and Evan~N. Feinberg and Joseph
 Gomes and Caleb Geniesse and Aneesh S. Pappu and Karl Leswing and Vijay Pande},
-title = {MoleculeNet: a benchmark for molecular machine learning},
+title = {aminoacid sequenceNet: a benchmark for molecular machine learning},
```
[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/peptides_hemolytic/transform.py:149]: Ooops, just an error, replaced molecule with aminoacid sequence

File: data/tabular/skin_reaction/meta.yaml (hunk 7 above)
Comment (AdrianM0) [reply data/tabular/skin_reaction/meta.yaml]: fixed

File: data/tabular/skin_reaction/meta.yaml (hunk 6 above)
Comment (AdrianM0) [reply data/tabular/skin_reaction/meta.yaml]: fixed

File: data/tabular/peptides_hemolytic/transform.py [hunk 17]
```diff
@@ -97,6 +97,7 @@ def get_and_transform_data():
                     {"verb": "kill red blood cells"},
```
[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/peptides_hemolytic/transform.py]: fixed

File: data/tabular/sr_are_tox21/transform.py (hunk 9 above)
Comment (AdrianM0) [reply data/tabular/sr_are_tox21/transform.py:70]: image

File: data/tabular/nr_ppar_gamma_tox21/transform.py [hunk 18]
```diff
@@ -63,20 +64,16 @@ def get_and_transform_data():
...
-                        "verb": "is active in the peroxisome proliferator-activated receptor gamma toxicity assay"
-                    },
                     {"adjective": "toxic in the NR-PPAR-gamma assay"},
                     {
                         "adjective": "toxic in the NR-peroxisome proliferator-activated receptor gamma assay"
                     },
                     {
                         "adjective": "toxic in the peroxisome proliferator-activated receptor gamma assay"
                     },
+                    {"gerund": "showing toxicity in the NR-PPAR-gamma assay"},
```
Comment (MicPie) [inline data/tabular/nr_ppar_gamma_tox21/transform.py:74]: Here you can but showing, displaying, and exhibiting like that into the text template:
`{#showing|displaying|exhibiting!}`
Like that you need only one gerund here.

File: data/tabular/potassium_ion_channel_kir2_1_butkiewicz/transform.py [hunk 19]
```diff
@@ -57,12 +57,12 @@ def get_and_transform_data():
                 "units": None,
                 "type": "boolean",
                 "names": [
-                    {"verb": "is blocking potassium ion channel activity"},
-                    {"verb": "blocks potassium ion channel activity"},
+                    {"gerund": "blocking of the potassium ion channel activity"},
```
Comment (MicPie) [inline data/tabular/potassium_ion_channel_kir2_1_butkiewicz/transform.py:60]:
```suggestion
                    {"gerund": "blocking the potassium ion channel"},
```

Comment (MicPie) [reply data/tabular/potassium_ion_channel_kir2_1_butkiewicz/transform.py:60]: Maybe use:
block channel
decrease activity
but "block activity" sounds strange, but maybe this is used in the literature?

File: data/tabular/sarscov2_vitro_touret/transform.py [hunk 20]
```diff
@@ -56,6 +56,9 @@ def get_and_transform_data():
                     {"adjective": "active against the Corona virus"},
                     {"adjective": "active against SARSCoV2"},
                     {"adjective": "active against COVID19"},
+                    {"gerund": "fighting the Corona virus"},
```
Comment (MicPie) [inline data/tabular/sarscov2_vitro_touret/transform.py:59]: "fighting" sounds a little bit like slang here, maybe there is a more formal word?

**Review by MicPie — CHANGES_REQUESTED** [review]
Looks very good! Thank you!
Just some minor things I found. :-)

File: data/tabular/sarscov2_vitro_touret/transform.py (hunk 20 above)
Comment (AdrianM0) [reply data/tabular/sarscov2_vitro_touret/transform.py:59]: countering?

File: data/tabular/nr_ppar_gamma_tox21/transform.py (hunk 18 above)
Comment (AdrianM0) [reply data/tabular/nr_ppar_gamma_tox21/transform.py:74]: you mean I can do something like this?
```suggestion
                    {"gerund": "{#showing|displaying|exhibiting!} toxicity in the NR-PPAR-gamma assay"},
```

Comment (MicPie) [reply data/tabular/nr_ppar_gamma_tox21/transform.py:74]: Ah, sorry, I was unclear, the `{#showing|displaying|exhibiting!}` would need to move to the text template and the gerund then would look like that:
`{"gerund": "toxicity in the NR-PPAR-gamma assay"},`

But this functionality of using the text part sampling is a good idea! We can think about it in the next iteration.

Comment (AdrianM0) [reply data/tabular/nr_ppar_gamma_tox21/transform.py:74]: then it is not a gerund and just a noun

File: data/tabular/nr_ahr_tox21/transform.py [hunk 21]
```diff
@@ -41,6 +41,7 @@ def get_and_transform_data():
 
     # save to csv
     fn_data_csv = "data_clean.csv"
+    df["toxicity_NR-AhR"] = df[f"toxicity_{target_subfolder}"].astype(int)
```
[the remark this answers is not in the document]
Comment (MicPie) [reply data/tabular/nr_ahr_tox21/transform.py:44]: Ah, in the other files it is hardcoded like that.

File: data/tabular/nr_er_lbd_tox21/transform.py [hunk 22]
```diff
@@ -41,6 +41,7 @@ def get_and_transform_data():
 
     # save to csv
     fn_data_csv = "data_clean.csv"
+    df["toxicity_NR-ER-LBD"] = df[f"toxicity_{target_subfolder}"].astype(int)
```
[the remark this answers is not in the document]
Comment (MicPie) [reply data/tabular/nr_er_lbd_tox21/transform.py:44]: as above

File: data/tabular/sarscov2_vitro_touret/transform.py [hunk 23]
```diff
@@ -56,9 +56,7 @@ def get_and_transform_data():
                     {"adjective": "active against the Corona virus"},
                     {"adjective": "active against SARSCoV2"},
                     {"adjective": "active against COVID19"},
-                    {"gerund": "fighting the Corona virus"},
-                    {"gerund": "fighting against SARSCoV2"},
-                    {"gerund": "fighting against COVID19"},
+                    {"gerund": "countering the effects of the Corona virus"},
```
[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/sarscov2_vitro_touret/transform.py:59]: thought at some point you said that acting against is not that good :D

File: data/tabular/sarscov2_vitro_touret/meta.yaml [hunk 24]
```diff
@@ -15,6 +15,7 @@ targets:
           - adjective: active against the Corona virus
           - adjective: active against SARSCoV2
           - adjective: active against COVID19
+          - gerund: countering the effects of the Corona virus
```
Comment (AdrianM0) [inline data/tabular/sarscov2_vitro_touret/meta.yaml:18]:
```suggestion
          - gerund: mitigating the effects of the Corona virus
```

File: data/tabular/sarscov2_vitro_touret/transform.py (hunk 23 above)
[the remark this answers is not in the document]
Comment (MicPie) [reply data/tabular/sarscov2_vitro_touret/transform.py]: huh, lol, I hope not, but then lets go for mitigating ;-)

File: data/tabular/nr_ppar_gamma_tox21/transform.py (hunk 18 above)
Comment (MicPie) [reply data/tabular/nr_ppar_gamma_tox21/transform.py:74]: yeah, I'm also fine with keeping it like that, in the end the result will be the same

**Review by MicPie — APPROVED** [review]

**Pull request merged**